### DISCUSSION TITLE
Reduce blank spacing and increase screen real estate efficiency

### DISCUSSION THREAD

**Discussion by abc91199** [microsoft/vscode-jupyter#9637] [New Ideas]
There are a lot of blank spaces/paddings/margins in and between cell blocks that affects readability and make it difficult to transverse through long codes with many cell blocks. See screencaps below for illustration.

Captured from vscode jupyter:
![vscode_jupyter](https://user-images.githubusercontent.com/16594734/162360061-2535e95e-1b80-4855-bf37-9dda1c8d247c.jpg)

Captured from jupyterlab in chrome:
![vanilla_jupyterlab_chrome](https://user-images.githubusercontent.com/16594734/162360068-373029e9-2836-44b6-815b-5e64b36bc472.jpg)
 
(from same screen, same resolution, same font size)

In jupyterlab you can almost fit 2 times as much information into one screen view compared to vscode.

Went over the settings in vscode but there doesn't seem to be any option to change the spacing/padding/margins.

**Comment by digitarald** [comment]
There is a font-size setting for notebooks but nothing that allows a more compact vertical rendering by reducing margins and line-height. Maybe a "compact" rendering could be part of the customizations or exposed via more configurations. Moving this to _Ideas_ to gather more perspectives.

**Comment by 8b8** [comment]
Given the general nature of the request I would also like to point to the nifty feature in Jupyter lab, that I miss in VSCode.
This enables the output to the right of the code cell, allowing for more efficient use of space with typical monitor aspect ratios.

https://github.com/jupyterlab/jupyterlab/pull/10648

**Reply by amunger** [reply]
thanks for the link to that discussion. @IanMatthewHuff also started a discussion around the idea here https://github.com/microsoft/vscode/issues/143244

**Comment by amunger** [comment]
vscode insiders now includes a `notebook: output line height` setting that should help with this.
